pkg/operator/controllers/storageaccounts/storageaccounts_test.go

```diff
@@ -0,0 +1,210 @@
+package storageaccounts
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+
+	mgmtstorage "github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-06-01/storage"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
+	imageregistryclient "github.com/openshift/client-go/imageregistry/clientset/versioned"
+	imageregistryfake "github.com/openshift/client-go/imageregistry/clientset/versioned/fake"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	mock_storage "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/storage"
+	mock_subnet "github.com/Azure/ARO-RP/pkg/util/mocks/subnet"
+	"github.com/Azure/ARO-RP/pkg/util/subnet"
+)
+
+var (
+	subscriptionId           = "0000000-0000-0000-0000-000000000000"
+	clusterResourceGroupName = "aro-iljrzb5a"
+	clusterResourceGroupId   = "/subscriptions/" + subscriptionId + "/resourcegroups/" + clusterResourceGroupName
+	vnetResourceGroup        = "vnet-rg"
+	vnetName                 = "vnet"
+	subnetNameWorker         = "worker"
+	subnetNameMaster         = "master"
+
+	storageSuffix              = "random-suffix"
+	clusterStorageAccountName  = "cluster" + storageSuffix
+	registryStorageAccountName = "image-registry-account"
+
+	resourceIdMaster = "/subscriptions/" + subscriptionId + "/resourceGroups/" + vnetResourceGroup + "/providers/Microsoft.Network/virtualNetworks/" + vnetName + "/subnets/" + subnetNameMaster
+	resourceIdWorker = "/subscriptions/" + subscriptionId + "/resourceGroups/" + vnetResourceGroup + "/providers/Microsoft.Network/virtualNetworks/" + vnetName + "/subnets/" + subnetNameWorker
+)
+
+func getValidClusterInstance() *arov1alpha1.Cluster {
+	return &arov1alpha1.Cluster{
+		Spec: arov1alpha1.ClusterSpec{
+			ClusterResourceGroupID: clusterResourceGroupId,
+			StorageSuffix:          storageSuffix,
+			OperatorFlags: arov1alpha1.OperatorFlags{
+				ENABLED: "true",
```

Comment:
We should test what happens when the controller is disabled -- i.e. that this flag does anything :)

Reply:
Added.